### PR TITLE
Bring ABC size down to 16

### PR DESCRIPTION
**Pull request by Drenmi**
I wanted to cement the excellent work done by @buehmann in #7450 by bringing the bar down to 16. That way we can't have new offenses creeping in without notice.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [X] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/node_pattern.rb [hunk 1]
```diff
@@ -429,6 +429,7 @@ def compile_ellipsis
         [0..Float::INFINITY, 'true']
       end
 
+      # rubocop:disable Metrics/AbcSize
```
Comment (Drenmi) [inline lib/rubocop/node_pattern.rb]: Couldn't find any way to refactor this method that didn't make it worse. Pretty sure this is as good as it gets with the current design, but open to suggestions.

**Comment by bbatsov** [comment]
🙇

File: lib/rubocop/node_pattern.rb (hunk 1 above)
Comment (buehmann) [reply lib/rubocop/node_pattern.rb]: Challenge accepted 😉 https://github.com/rubocop-hq/rubocop/pull/7457

File: lib/rubocop/cop/metrics/utils/abc_size_calculator.rb [hunk 2]
```diff
@@ -55,11 +54,26 @@ def evaluate_branch_nodes(node)
...
           end
 
-          def node_has_else_branch?(node)
+          def evaluate_condition_node(node)
+            @condition += 1 if else_branch?(node)
+            @condition += 1
+          end
+
+          def else_branch?(node)
             %i[case if].include?(node.type) &&
```
Comment (Drenmi) [inline lib/rubocop/cop/metrics/utils/abc_size_calculator.rb]: @jonas054 I suspect we could have a subtle bug here. We check for `#else_branch?` on `CONDITION_NODES`, which comes from `CyclomaticComplexity::COUNTED_NODES`, but that list does not include `case`:

```ruby
COUNTED_NODES = %i[if while until for
                   rescue when and or].freeze
```

**Comment by jaredbeck** [comment]
> Reduce ABC size of AbcSizeCalculator (5eb411b)

so meta

**Pull request merged**